cmd/restic/cmd_forget.go

```diff
@@ -138,14 +138,26 @@ func init() {
 func verifyForgetOptions(opts *ForgetOptions) error {
 	if opts.Last < -1 || opts.Hourly < -1 || opts.Daily < -1 || opts.Weekly < -1 ||
 		opts.Monthly < -1 || opts.Yearly < -1 {
+		// TODO: this seem to have no effect. there is a different error message showing up, e.g. for --keep-last -5
 		return errors.Fatal("negative values other than -1 are not allowed for --keep-*")
 	}
 
+	var allKeepWithinXUndefined = true
 	for _, d := range []restic.Duration{opts.Within, opts.WithinHourly, opts.WithinDaily,
 		opts.WithinMonthly, opts.WithinWeekly, opts.WithinYearly} {
 		if d.Hours < 0 || d.Days < 0 || d.Months < 0 || d.Years < 0 {
 			return errors.Fatal("durations containing negative values are not allowed for --keep-within*")
 		}
+		if d.Hours != 0 || d.Days != 0 || d.Months != 0 || d.Years != 0 {
```

Comment:
@arberg regarding 1.: these lines should be correct. `d` is a `Duration` which is defined as
```
type Duration struct {
	Hours, Days, Months, Years int
}
```
The duration specification has no week.

Reply:
... and each `opts.Within*` option has such a duration specification as argument. That's my understanding.